tests/group_functions/group_scan.h

```diff
@@ -20,11 +20,149 @@
 
 #include <valarray>
 
+#include "../../util/accuracy.h"
 #include "group_functions_common.h"
 
-template <int D, typename T, typename U>
+template <int D, typename T, typename U, typename Group, bool with_init,
+          typename I, typename OpT>
 class joint_scan_group_kernel;
 
+constexpr int init = 42;
+
+template <typename Group, int D>
+Group get_group(const sycl::nd_item<D>& item) {
+  if constexpr (std::is_same_v<std::decay_t<Group>, sycl::sub_group>)
+    return item.get_sub_group();
+  else
+    return item.get_group();
+}
+
+template <bool with_init, typename I, typename T, typename U, typename Group,
+          typename OpT>
+auto joint_inclusive_scan_helper(Group group, T* v_begin, T* v_end,
+                                 U* r_i_begin, OpT op) {
+  if constexpr (with_init) {
+    return sycl::joint_inclusive_scan(group, v_begin, v_end, r_i_begin, op,
+                                      I(init));
+  } else
+    return sycl::joint_inclusive_scan(group, v_begin, v_end, r_i_begin, op);
+}
+
+template <bool with_init, typename I, typename T, typename U, typename Group,
+          typename OpT>
+auto joint_exclusive_scan_helper(Group group, T* v_begin, T* v_end,
+                                 U* r_e_begin, OpT op) {
+  if constexpr (with_init) {
+    return sycl::joint_exclusive_scan(group, v_begin, v_end, r_e_begin, I(init),
+                                      op);
+  } else
+    return sycl::joint_exclusive_scan(group, v_begin, v_end, r_e_begin, op);
+}
+
+template <int D, typename T, typename U, typename Group, bool with_init,
+          typename I = U, typename OpT>
+void check_scan(sycl::queue& queue, size_t size,
+                sycl::nd_range<D> executionRange, OpT op) {
+  std::vector<T> v(size);
+  std::iota(v.begin(), v.end(), T(1));
```

Comment:
I suggest we fill input buffer with random numbers and print a seed to reproduce the input.